packages/babel-plugin/src/utils/minify.js

```diff
@@ -5,18 +5,10 @@ const haveSameLocation = (element1, element2) => {
   return element1.line === element2.line && element1.column === element2.column
 }
 
-const isAutoInsertedRule = element => {
-  if (element.type !== 'rule' || !element.parent) {
-    return false
-  }
-
-  let parent = element
-  do {
-    parent = parent.parent
-  } while (parent && parent.type !== 'rule')
```

Comment:
No idea why I've included this loop in the past - all tests pass without it. We now directly compare locations of `element` and `element.parent`.

The issue was that we temporarily replace interpolations with strings that shouldn't appear in real styles (can be seen [here](https://github.com/emotion-js/emotion/blob/ae0f650b5ad8ef658b61edca6157fe208be23b43/packages/babel-plugin/src/utils/minify.js#L133-L149)). Such output string is being processed by Stylis (that's why we replace them in the first place, so we can compile styles with Stylis).So Stylis never knows about our little hack and it could, accidentally, treat our string as part of the selector for a rule.

This was happening for input like this:
```js
  ${fullWidth}

  button {
```
Since it was transformed by us to:
```
  xxx0:xxx

  button {
```

And if we had an at-rule within such block then for a rule in it we didn't recognize a rule to be auto-inserted. This information is being utilized by [`toInputTree`](https://github.com/emotion-js/emotion/blob/ae0f650b5ad8ef658b61edca6157fe208be23b43/packages/babel-plugin/src/utils/minify.js#L21) which tries to convert the AST back to a shape representing the input string to make it possible to print it back to as a minified string.